### PR TITLE
migrate service tests to GH actions

### PR DESCRIPTION
**Pull request by chris48s**
Finally closes https://github.com/badges/shields/issues/8109
(although we still have the daily-tests repo to do and a bit more work on service tests that require credentials :grimacing: )

The core purpose of this is moving the services build to run on GitHub Actions instead of Circle CI, but there are some intricacies.

### CODE REVIEW

**Comment by chris48s** [comment]
This is 90% done, but marking it as 'on hold' for the moment while we sort out the other 10%

**Comment by chris48s** [comment]
Bleh. Unexpectedly, it looks like the token we get in CI can't call the API for gists or file contents. Need to think about how to deal with that. Also (more predictably) running ~250 github tests on 2 node versions lots of times is enough to exceed the rate limit.

**Comment by chris48s** [comment]
TODO: use a PAT if the PR is from a local branch (like the integration tests)

File: .circleci/config.yml
Comment (chris48s) [inline .circleci/config.yml:1]: Deleting the remaining circle config causes the Circle CI build to fail with "No `.circleci/config.yml` was found in your project. Add a configuration file to your repository or refer to documentation.". When we get to merging this I will disable Circle from trying to run on the repository but for now I am going to leave it while there are still other open PRs running the service tests on Circle. Also I will update the branch protection rules, delete all the env vars from Circle, etc.

File: .github/workflows/test-services.yml
Comment (chris48s) [inline .github/workflows/test-services.yml:38]: If we submit a PR from a branch on this repo (Anyone with push access can do this. Also importantly, this applies to dependabot PRs - they're subimtted from a local branch, not a fork), all of the service tests can run using credentials which I have configured as secrets.

If we recieve a PR from a fork (like this one) tests for the following services will fail due to lack of credentials, or be flaky due to reduced rate limit:
- Libraries.io
- OBS
- Symfony Insight
- Twitch
- Wheelmap
- Youtube
- GitHub gists only, but not the rest of GitHub (more on this further down)

All other service tests that do not require auth will run fine in either case.
As discussed on the call, this means this are a small handful of services where if an external contributor wants to submit a PR changing them, we'll have to verify the tests locally and then edit the PR title before merging.
Seperately, I will have a look at finding a way to run tests for these services in another context.

You can see an example of service tests running with the required credentials on a PR from a local branch at https://github.com/badges/shields/actions/runs/3923463559/jobs/6707091330

File: services/github/gist/github-gist-last-commit-redirect.tester.js
Comment (chris48s) [inline services/github/gist/github-gist-last-commit-redirect.tester.js:4]: Annoyingly, the built-in `GITHUB_TOKEN` secret has enough permissions to run _neeeeearly_ all of the GitHub test suite. The one exception to this is Gists. Also it is not possible to grant a token a `gist: read` scope: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token This is a limitation which does apply to github actions/workflow tokens but doesn't exist for PATs.

The way I've proposed to deal with this is we rename the `GithubGist` service to just `Gist`. This will mean if I submit a PR

- "do a thing to [github] services" then all the tests should run and pass, even if the PR is submitted from a fork
- "do a thing to [gist] services" then all the tests should run and pass from a local branch but will fail from a fork

The downside of that is we have to remember that `GithubGist` has a different service tag from `Github`. The other thing we could do is leave it and accept that a handful of tests will fail if we try to run the full github suite from a forked PR.

File: .github/workflows/test-services-17.yml
Comment (chris48s) [inline .github/workflows/test-services-17.yml:34]: I've prefixed these just to help organise things a bit.

File: .github/workflows/test-services.yml
Comment (chris48s) [reply .github/workflows/test-services.yml:38]: here's an equivalent build trying to run the same service tests from a fork
https://github.com/badges/shields/actions/runs/3923735407

**Comment by chris48s** [comment]
@calebcartwright - this one is now ready for a review

File: core/service-test-runner/pull-request-services-cli.js
Comment (calebcartwright) [inline core/service-test-runner/pull-request-services-cli.js:20]: I know this is right because it works, but I'm having a little trouble convincing myself I've got the correct mental model of the flow. Is it just that this is expecting that the `...:prepare` npm script was invoked with the title arg in the first true arg position?

**Review by calebcartwright — APPROVED** [review]
Works for me, and definitely aligns to the plan we came up with at the last team meeting. Understand the need and agreed with the approach for special-case naming the GH gist badges as well. I'm fine with the name as-is, but suppose one thing worth considering is using some kind of additional (non-`Github`) identifier to reduce the risk of any name collisions down the road, e.g. `GHGist` or something like that.

Happy to proceed as-is, happy to re-:+1: if you decide to make any additional tweaks

File: core/service-test-runner/pull-request-services-cli.js
Comment (chris48s) [reply core/service-test-runner/pull-request-services-cli.js:20]: Even though we invoke it with `npm run...`,
`argv[0]` is the full path to `node`
`argv[1]` is the full path to `pull-request-services-cli.js`
and then `argv[2]` is the first positional argument.

Comment (chris48s) [reply core/service-test-runner/pull-request-services-cli.js:20]: and we call it here
https://github.com/badges/shields/blob/00fe69418eeeb963ca94f817438a1a744bc3b355/.github/actions/service-tests/action.yml#L47-L57

**Pull request merged**
